dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/models/SubSnpCoreFields.java

```diff
@@ -306,17 +348,25 @@ private String getAlleleInForwardStrand(String allele, Orientation orientation)
      * - rs10721689 :  "alleles" are reverse when orientations are 1 -1 1, forward when 1 -1 -1
      */
     public String getAllelesInForwardStrand() {
-        boolean forward = this.getSubSnpOrientation().equals(Orientation.FORWARD)
-                ^ this.getSnpOrientation().equals(Orientation.FORWARD)
-                ^ this.getContigOrientation().equals(Orientation.FORWARD);
+        return getForwardOrientedAlleles(getAlleleOrientation(), this.getAlleles());
+    }
 
-        String alleles = this.getAlleles();
-        String forwardAlleles = forward ? alleles : calculateReverseComplement(alleles);
+    private String getForwardOrientedAllele(boolean forward, String allele) {
+        return getTrimmedAllele(forward ? allele : calculateReverseComplement(allele));
+    }
 
+    private String getForwardOrientedAlleles(boolean forward, String alleles) {
+        String forwardAlleles = forward ? alleles : calculateReverseComplement(alleles);
         String[] splitAlleles = forwardAlleles.split("/", -1);
         return Stream.of(splitAlleles).map(this::getTrimmedAllele).collect(Collectors.joining("/"));
     }
 
+    private boolean getAlleleOrientation() {
+        return this.getSubSnpOrientation().equals(Orientation.FORWARD)
```

Comment:
I suggest adding to the tests some variants covering different orientation combinations (if we don' t already have them)

Reply:
this code was already in there. tests should be here: https://github.com/EBIvariation/eva-tools/blob/543fe7f1aabd321e6de23997873f02d9207eb90d/dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/models/SubSnpCoreFieldsForwardStrandMappingTest.java#L419